akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowFoldAsyncSpec.scala

```diff
@@ -0,0 +1,259 @@
+/**
+ * Copyright (C) 2014-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import scala.util.control.NoStackTrace
+
+import scala.concurrent.{ Await, Future }
+import scala.concurrent.duration._
+
+import akka.NotUsed
+import akka.testkit.{ AkkaSpec, TestLatch }
+import akka.stream.ActorMaterializer
+import akka.stream.ActorAttributes.supervisionStrategy
+import akka.stream.Supervision.{ restartingDecider, resumingDecider }
+import akka.stream.impl.ReactiveStreamsCompliance
+import akka.stream.testkit._, Utils._
+
+class FlowFoldAsyncSpec extends StreamSpec {
+  implicit val materializer = ActorMaterializer()
+  implicit def ec = materializer.executionContext
+
+  "A FoldAsync" must {
+    val input = 1 to 100
+    val expected = input.sum
+    val inputSource = Source(input).filter(_ ⇒ true).map(identity)
+    val foldSource = inputSource.foldAsync[Int](0) { (a, b) ⇒
+      Future(a + b)
+    }.filter(_ ⇒ true).map(identity)
```

Comment:
why the filter+map(identity)? If the intention was to exercise concurrency then this will not do it as everything is fused. Put an `.async` at the end and it will force an async boundary which is more likely what you want. I am not sure it is needed though.